### DISCUSSION TITLE
Attributes of groups and enumerations instead of primitive datatypes

### DISCUSSION THREAD

**Discussion by flappah** [LiorBanai/HDF5-CSharp#265] [Q&A]
Hi Lior,

First off, Happy New Year!

Question, there's been a revision to the IHO S111 standard that requires me to use enumerations of some attributes on groups. Now I know how to define attributes of primitive datatypes (short, int, string, etc) but how do I define attributes of an enumerated datatype? There is for example this attribute with the name commonPointRule of the SurfaceCurrent group. It has the following values: 

1: average,
2: low,
3: high,
4: all

Now with the previous S111 standards I could just define it is being of type byte. Now with the current it requires me to be of the above mentioned enumeration. The HDF viewer then 'knows' which possible enumerated values it can expect. Is there any way to create those attributes with the HDF5-CSharp library?

Thanks!

**Comment by flappah** [comment]
Quick update, I managed to circumvent the issue by using the lower lying HDF5 library your HDF5-CSharp library is wrapped around. It does require me to use the PInvokes quite extensively and really take care of the GCHandle.Alloc's and Free's but I can mix calls to your library and also use the enumerated types when creating attributes. I can mix calls to your library with calls to the underlying HDF5 PInvoked C-library so there is no immediate need for my above mentioned request. 

To give some insights on how I used it I've added this test project to Github. You can find it at https://github.com/flappah/Hdf5Fase2Test

**Reply by LiorBanai** [reply]
@flappah Thanks a lot. I'll look at it. I'm currently  quite busy but I will get to it.
